sklearn/utils/_repr_html/tests/test_params.py

```diff
@@ -69,6 +76,143 @@ def test_read_params():
+        r'\s*<a class="param-doc-link"'
+        r'\s*rel="noreferrer" target="_blank"'
+        r'\shref="mock_module\.MockEstimator\.html#:~:text=a,-int">'
+        r"\s*a"
+        r'\s*<span class="param-doc-description">a: int<br><br>'
+        r"Description of a\.</span>"
+        r"\s*</a>"
+        r"\s*</td>"
+    )
+    assert re.search(html_param_a, html_output, flags=re.DOTALL)
+    html_param_b = (
+        r'<td class="param">'
+        r'.*<a class="param-doc-link"'
+        r'\s*rel="noreferrer" target="_blank"'
+        r'\shref="mock_module\.MockEstimator\.html#:~:text=b,-str">'
+        r"\s*b"
+        r'\s*<span class="param-doc-description">b: str<br><br></span>'
+        r"\s*</a>"
+        r"\s*</td>"
+    )
+    assert re.search(html_param_b, html_output, flags=re.DOTALL)
+
+
+def test_params_html_repr_without_doc_links():
+    """Test `_params_html_repr` when `link_to_param_doc` returns None."""
+
+    class MockEstimatorWithoutDoc:
+        __module__ = "sklearn.mock_module"
+        __qualname__ = "MockEstimatorWithoutDoc"
+        # No docstring defined on this test class.
+
+    params = ParamsDict(
+        params={"a": 1, "b": "value"},
+        non_default=("a",),
+        estimator_class=MockEstimatorWithoutDoc,
+    )
+    html_output = _params_html_repr(params)
+    # Check that no doc links are generated
+    assert "?" not in html_output
+    assert "Click to access" not in html_output
+    html_param_a = (
+        r'<td class="param">a</td>'
+        r'\s*<td class="value">1</td>'
+    )
+    assert re.search(html_param_a, html_output, flags=re.DOTALL)
+    html_param_b = (
+        r'<td class="param">b</td>'
+        r'\s*<td class="value">&#x27;value&#x27;</td>'
+    )
+    assert re.search(html_param_b, html_output, flags=re.DOTALL)
+
+
+def test_generate_link_to_param_doc_basic():
+    class MockEstimator:
+        """Mock class.
+
+        Parameters
+        ----------
+        alpha : float
+            Regularization strength.
+        beta : int
+            Some integer parameter.
+        """
+
+    doc_link = "mock_module.MockEstimator.html"
+    url = _generate_link_to_param_doc(MockEstimator, "alpha", doc_link)
+    assert url == "mock_module.MockEstimator.html#:~:text=alpha,-float"
+
+    url = _generate_link_to_param_doc(MockEstimator, "beta", doc_link)
+    assert url == "mock_module.MockEstimator.html#:~:text=beta,-int"
+
+
+def test_generate_link_to_param_doc_param_not_found():
+    class MockEstimator:
+        """Mock class
+
+        Parameters
+        ----------
+        alpha : float
+            Regularization strength.
+        """
+
+    doc_link = "mock_module.MockEstimator.html"
+    url = _generate_link_to_param_doc(MockEstimator, "gamma", doc_link)
+
+    assert url is None
+
+
+def test_generate_link_to_param_doc_special_characters():
+    class MockEstimator:
+        """Mock class
+        param with space : {'stay', 'leave'} default='don't know'
```

Comment:
does such a param exists ? I don't see how a parameter could have a space in it's name in Python.